restler/engine/transport_layer/response.py

```diff
@@ -115,7 +117,19 @@ def has_bug_code(self):
 
         """
         if self._status_code:
-            return self._status_code.startswith('5')
+            if Settings().custom_non_bug_codes:
+                # If the non-bug-code list is nonempty,
+                # return False only if the status code exists in the list.
+                for code in Settings().custom_non_bug_codes:
```

Comment:
suggest rephrasing 'All codes except the ones in that list should be flagged as bugs.  Hence, return False ...'